test/e2e/source/dataplane_test.go

```diff
@@ -0,0 +1,268 @@
+// +build e2e_ginkgo
+
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package source
+
+import (
+	"context"
+	"encoding/json"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	. "github.com/cloudevents/sdk-go/v2/test"
+	cetypes "github.com/cloudevents/sdk-go/v2/types"
+
+	"knative.dev/eventing/test/lib/naming"
+	"knative.dev/eventing/test/lib/recordevents"
+	rrecordevents "knative.dev/eventing/test/lib/resources/recordevents"
+
+	sourcesv1beta1 "knative.dev/eventing-kafka/pkg/apis/sources/v1beta1"
+	"knative.dev/eventing-kafka/test/e2e/helpers"
+	kafkasource "knative.dev/eventing-kafka/test/lib/resources/kafkasource/v1beta1"
+	kafkatopic "knative.dev/eventing-kafka/test/lib/resources/kafkatopic/v1beta2"
+)
+
+type auth struct {
+	bootstrapServer string
+	SASLEnabled     bool
+	TLSEnabled      bool
+}
+
+type kafkaMessage struct {
+	Key         string
+	Headers     map[string]string
+	Payload     string
+	ContentType string
+	CloudEvent  bool
+}
+
+const (
+	kafkaBootstrapUrlPlain = "my-cluster-kafka-bootstrap.kafka.svc:9092"
+	kafkaBootstrapUrlTLS   = "my-cluster-kafka-bootstrap.kafka.svc:9093"
+	kafkaBootstrapUrlSASL  = "my-cluster-kafka-bootstrap.kafka.svc:9094"
+
+	kafkaClusterName      = "my-cluster"
+	kafkaClusterNamespace = "kafka"
+)
+
+var (
+	ts, _ = cetypes.ParseTime("2018-04-05T17:31:00Z")
```

Comment:
Should this be a dynamically generated value (recent... few seconds ago) ?

Reply:
why not?